fineract-provider/src/main/java/org/apache/fineract/portfolio/charge/domain/Charge.java

```diff
@@ -621,4 +600,34 @@ public TaxGroup getTaxGroup() {
     public void setTaxGroup(TaxGroup taxGroup) {
         this.taxGroup = taxGroup;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof Charge)) return false;
+        Charge charge = (Charge) o;
+        return penalty == charge.penalty &&
+                active == charge.active &&
+                deleted == charge.deleted &&
+                name.equals(charge.name) &&
+                amount.equals(charge.amount) &&
+                currencyCode.equals(charge.currencyCode) &&
+                chargeAppliesTo.equals(charge.chargeAppliesTo) &&
+                chargeTimeType.equals(charge.chargeTimeType) &&
+                chargeCalculation.equals(charge.chargeCalculation) &&
+                chargePaymentMode.equals(charge.chargePaymentMode) &&
```

Comment:
use Objects.equals helper here, as some of these may well be null, and would cause NPE as is.